app/views/users/_table.html.erb

```diff
@@ -71,7 +71,9 @@
         <td><%= maybe_link_user(link_to_students, user) %></td>
         <td><%= staff.detect{|u| u.user_id == user.id}.role.humanize %></td>
         <td><%= if user.id != current_user.id
-                link_to 'Impersonate', impersonate_user_path(user), method: 'post'
+                  if current_user.site_admin? || current_user.professor_ever?
```

Comment:
This check has already been done, actually, and is cached as the variable `link_to_students` on line 1 of this file.  (Since making the `professor_ever?` check requires a database query, I try to avoid doing it repeatedly if it's not needed...)

Reply:
Good point.